scripts/configure.js

```diff
@@ -0,0 +1,151 @@
+const _ = require('lodash');
+const inquirer = require('inquirer');
+const winston = require('winston');
+const request = require('request');
+const fs = require('fs');
+const path = require('path');
+
+const configFilePath = path.resolve(__dirname, '../config/core.json');
+
+const log = new winston.Logger({
+    transports: [new winston.transports.Console({
+        level: 'info',
+        handleExceptions: true,
+        colorize: true,
+    })],
+});
+
+const regex = {
+    fqdn: new RegExp(/^https?:\/\/(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,64}\/?$/),
+    ipv4: new RegExp(/((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.|$)){4}/),
+};
+
+const argv = require('yargs')
+.usage('npm run configure -- <arguments>')
+    .describe('p', 'The panel url to pull the configuration from')
+    .alias('p', 'panel-url')
+    .describe('t', 'The configuration token')
+    .alias('t', 'token')
+    .boolean('o')
+    .describe('o', 'Overwrite existing configuration file')
+    .alias('o', 'overwrite')
+    .alias('h', 'help')
+    .help('h').argv;
+
+const params = {
+    panelurl: {
+        name: 'Panel url',
+        value: argv.p,
+        validate: value => {
+            if (value === '') {
+                return 'The panel url is required';
+            }
+            if (!value.match(regex.fqdn) && !value.match(regex.ipv4)) {
+                return 'Please provide either a URL (with protocol) or an IPv4 address.';
+            }
+            return true;
+        },
+    },
+    token: {
+        name: 'Configuration token',
+        value: argv.t,
+        validate: value => {
+            if (value === '') {
+                return 'A configuration token is required.';
+            }
+            if (value.length !== 32) {
+                return 'The token provided is invalid. It must be 32 characters long.';
+            }
+            return true;
+        },
+    },
+    overwrite: {
+        name: 'Overwrite',
+        value: argv.o,
+    },
+};
+
+// Validate and filter the provided command line params
+function validateParams() {
```

Comment:
The params should be validated with yargs as it's possible i think. Will change that